doc/filters/country_name.rst

```diff
@@ -16,6 +16,32 @@ By default, the filter uses the current locale. You can pass it explicitly:
     {# États-Unis #}
     {{ 'US'|country_name('fr') }}
 
+The locale can contain more than two letters depending on the region:
+
+.. code-block:: twig
+
+    {# 美國 #}
+    {{ 'US'|country_name('zh_Hant_HK') }}
+
+.. note::
+
+    For more information on the format of the locale:
+
+    See https://www.rfc-editor.org/info/bcp47 for the specifications.
+
+    It is documented by https://www.php.net/manual/en/class.locale.php.
+
+    Taking the previous example into consideration, the filter searches for the 'US' country key within the array located
+    in the 'zh_Hant_HK.php' file corresponding to the locale and then retrieves the associated country name.
```

Comment:
this sentence does not make sense. It talks about an `zh_Hant_HK.php` file that won't be known by the user reading this documentation (and this explanation was wrong anyway, as it was missing all the cases about how locales are using fallback data as part of the processing).
I suggest removing this sentence entirely. We don't need to describe the internal implementation here (especially given that it does not describe the implementation of the filter but tries to describes the internal implementation of another package used in the filter)